### PR TITLE
Bindings for Terminal endpoints

### PR DESCRIPTION
**Pull request by daz-stripe**
Bindings for Terminal endpoints
- Connection Tokens
- Locations
- Readers

### CODE REVIEW

File: tests/Stripe/Terminal/ConnectionTokenTest.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+class ConnectionTokenTest extends \Stripe\TestCase
+{
+    public function testIsCreatable()
+    {
+      $this->expectsRequest(
+          'post',
+          '/v1/terminal/connection_tokens'
+      );
+      $resource = ConnectionToken::create();
+      $this->assertInstanceOf("Stripe\\Terminal\\ConnectionToken", $resource);
```
Comment (ob-stripe) [inline tests/Stripe/Terminal/ConnectionTokenTest.php]: PHP uses 4 spaces for indents, but you only used only 2 spaces here.

File: tests/Stripe/Terminal/LocationTest.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+          "display_name" => "name",
+          "address" => [
+              "line1" => "line1",
+              "country" => "US",
+              "state" => "CA",
+              "postal_code" => "12345",
+              "city" => "San Francisco"
+          ]
+      ]);
+      $this->assertInstanceOf("Stripe\\Terminal\\Location", $resource);
```
Comment (ob-stripe) [inline tests/Stripe/Terminal/LocationTest.php]: Same indentation issue in this function.

File: tests/Stripe/Terminal/ReaderTest.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    public function testIsCreatable()
+    {
+      $this->expectsRequest(
+          'post',
+          '/v1/terminal/readers',
+          ["registration_code" => "a-b-c"]
+      );
+      $resource = Reader::create(['registration_code' => 'a-b-c']);
+      $this->assertInstanceOf("Stripe\\Terminal\\Reader", $resource);
```
Comment (ob-stripe) [inline tests/Stripe/Terminal/ReaderTest.php]: And here too.

File: lib/Terminal/ConnectionToken.php [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ * @property string $connection_token
+ *
+ * @package Stripe\Terminal
+ */
+class ConnectionToken extends \Stripe\ApiResource
+{
+    const OBJECT_NAME = "terminal.connection_token";
+
+    use \Stripe\ApiOperations\Create;
+}
```
Comment (ob-stripe) [inline lib/Terminal/ConnectionToken.php]: Need to add a newline character at end of file to satisfy the linter.

File: lib/Terminal/Location.php [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ */
+class Location extends \Stripe\ApiResource
+{
+    const OBJECT_NAME = "terminal.location";
+
+    use \Stripe\ApiOperations\All;
+    use \Stripe\ApiOperations\Create;
+    use \Stripe\ApiOperations\Retrieve;
+    use \Stripe\ApiOperations\Update;
+}
```
Comment (ob-stripe) [inline lib/Terminal/Location.php]: Same here.

File: lib/Terminal/Reader.php [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+ */
+class Reader extends \Stripe\ApiResource
+{
+    const OBJECT_NAME = "terminal.reader";
+
+    use \Stripe\ApiOperations\All;
+    use \Stripe\ApiOperations\Create;
+    use \Stripe\ApiOperations\Retrieve;
+    use \Stripe\ApiOperations\Update;
+}
```
Comment (ob-stripe) [inline lib/Terminal/Reader.php]: And here.

File: lib/Terminal/ConnectionToken.php [hunk 7]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Stripe\Terminal;
+
+/**
+ * Class ConnectionToken
+ *
+ * @property string $connection_token
```
Comment (brandur-stripe) [inline lib/Terminal/ConnectionToken.php]: Minor, but can we change this to the new `$secret` property?

**Comment by brandur-stripe** [comment]
ptal @daz-stripe

File: lib/Terminal/ConnectionToken.php (hunk 7 above)
Comment (daz-stripe) [reply lib/Terminal/ConnectionToken.php]: Yep!

**Comment by daz-stripe** [comment]
Updated and fixed the misname!
r? @brandur-stripe

**Comment by brandur-stripe** [comment]
Thanks!

**Comment by brandur-stripe** [comment]
Released as 6.19.0.

**Pull request merged**